ext/test/w3c_tracecontext_test/main.cc

```diff
@@ -207,16 +221,28 @@ int main(int argc, char *argv[])
         return 0;
       }};
 
+  testing::HttpRequestCallback stop_cb{
+      [&](testing::HttpRequest const & /*req*/, testing::HttpResponse &resp) {
+        std::cout << "Received request to stop server \n";
+        stop_server.store(true);
+        resp.code = 200;
+        return 0;
+      }};
+
   server["/test"] = test_cb;
+  server["/stop"] = stop_cb;
 
   // Start server
   server.start();
 
   std::cout << "Listening at http://" << default_host << ":" << port << "/test\n";
 
-  // Wait for console input
-  std::cin.get();
-
-  // Stop server
+  // Wait for signal to stop server
+  while (!stop_server.load())
+  {
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+  }
+  // received signal to stop server
+  std::cout << "Stopping server \n";
```

Comment:
update the [README.md](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/ext/test/w3c_tracecontext_test/README.md#usage) - step 4 if we now expect user to invoke `/stop` to stop the server.